<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>iSEEing is believing: the iSEE package for efficient, interactive, and reproducible exploration of data • iSEEdemo</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="iSEEing is believing: the iSEE package for efficient, interactive, and reproducible exploration of data">
<meta property="og:description" content="iSEEdemo">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">iSEEdemo</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/iSEEdemo.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/iSEE/iSEEdemo" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>iSEEing is believing: the iSEE package for
efficient, interactive, and reproducible exploration of data</h1>
                        <h4 data-toc-skip class="author">Federico
Marini<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>,
Charlotte Soneson<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>, Kevin Rue-Albrecht<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>
</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/iSEE/iSEEdemo/blob/HEAD/vignettes/iSEEdemo.Rmd" class="external-link"><code>vignettes/iSEEdemo.Rmd</code></a></small>
      <div class="hidden name"><code>iSEEdemo.Rmd</code></div>

    </div>

    
    
<p><img src="iSEE.png" align="right" alt="" width="120"></p>
<div class="section level2">
<h2 id="overview-of-the-workshop">Overview of the workshop<a class="anchor" aria-label="anchor" href="#overview-of-the-workshop"></a>
</h2>
<div class="section level3">
<h3 id="description">Description<a class="anchor" aria-label="anchor" href="#description"></a>
</h3>
<p>The iSEE package is a flexible, powerful and extendible application
to explore any dataset stored in SummarizedExperiment objects, including
single cell and spatially resolved data.</p>
<p>iSEE enables a multitude of panels to gain in-depth insight in your
data; it does so combining interactivity and reproducibility, and
providing an ideal companion to CELLxGENE for deeper dives into
data.</p>
<p>This package demo provides an overview of the functionality of the
<em><a href="https://bioconductor.org/packages/3.18/iSEE" class="external-link">iSEE</a></em>
Bioconductor package, and is intended as an appetizer to more.</p>
</div>
<div class="section level3">
<h3 id="pre-requisites">Pre-requisites<a class="anchor" aria-label="anchor" href="#pre-requisites"></a>
</h3>
<ul>
<li>Familiarity with single-cell sequencing data (ideally
scRNA-seq)</li>
<li>Familiarity with the SummarizedExperiment/SingleCellExperiment
class, or alternatively the Seurat or AnnData frameworks - basically,
any of the widely used ecosystem for working with/storing/processing
single cell data</li>
<li>Familiarity with single-cell data visualizations and
representations</li>
</ul>
</div>
<div class="section level3">
<h3 id="participation">Participation<a class="anchor" aria-label="anchor" href="#participation"></a>
</h3>
<p>Participants will be able to follow along in the live coding parts of
the demo.</p>
</div>
<div class="section level3">
<h3 id="r-bioconductor-packages-used">
<em>R</em> / <em>Bioconductor</em> packages used<a class="anchor" aria-label="anchor" href="#r-bioconductor-packages-used"></a>
</h3>
<ul>
<li>
<code>iSEE</code>: <a href="https://bioconductor.org/packages/iSEE" class="external-link uri">https://bioconductor.org/packages/iSEE</a>
</li>
<li>
<code>iSEEde</code>: <a href="https://bioconductor.org/packages/iSEEde" class="external-link uri">https://bioconductor.org/packages/iSEEde</a>
</li>
<li>
<code>iSEEindex</code>: <a href="https://bioconductor.org/packages/iSEEindex" class="external-link uri">https://bioconductor.org/packages/iSEEindex</a>
</li>
<li>
<code>iSEEpathways</code>: <a href="https://bioconductor.org/packages/iSEEpathways" class="external-link uri">https://bioconductor.org/packages/iSEEpathways</a>
</li>
<li>
<code>iSEEhub</code>: <a href="https://bioconductor.org/packages/iSEEhub" class="external-link uri">https://bioconductor.org/packages/iSEEhub</a>
</li>
</ul>
<!-- ## Time outline --><!-- | Activity                                        | Time | --><!-- |-------------------------------------------------|------| --><!-- | Overview of iSEE                                | 20m  | --><!-- | Building up your own iSEE instance              | 10m  | --><!-- | New and extra features in the iSEE ecosystem    | 15m  | -->
</div>
<div class="section level3">
<h3 id="workshop-goals-and-objectives">Workshop goals and objectives<a class="anchor" aria-label="anchor" href="#workshop-goals-and-objectives"></a>
</h3>
<div class="section level4">
<h4 id="learning-goals">Learning goals<a class="anchor" aria-label="anchor" href="#learning-goals"></a>
</h4>
<ul>
<li>Outline the features of iSEE, built upon the RStudio Shiny
framework</li>
<li>Utilize interactive GUI components and layouts to efficiently
extract information from biological data sets</li>
<li>Describe how to construct interactive apps and custom panels</li>
<li>Demonstrate how these package can smoothly be integrated into and
enhance existing workflows for the datasets at hand</li>
</ul>
</div>
<div class="section level4">
<h4 id="learning-objectives">Learning objectives<a class="anchor" aria-label="anchor" href="#learning-objectives"></a>
</h4>
<ul>
<li>Set up a local environment for running iSEE apps</li>
<li>Interact with components of the iSEE user interface to visually
inspect and discuss various data sets</li>
<li>Identify and locate configurable aspects of iSEE apps</li>
<li>Practice interactive visualization over a single-cell RNA-sequencing
workflow</li>
<li>Imagine use cases and future developments for interactive
visualization as part of computational workflows</li>
<li>Design and create iSEE instances for collections of datasets</li>
</ul>
</div>
</div>
</div>
<div class="section level2">
<h2 id="installing-and-loading-the-required-packages">Installing and loading the required packages<a class="anchor" aria-label="anchor" href="#installing-and-loading-the-required-packages"></a>
</h2>
<p>To run the content presented in this demo, run this following chunk
in your R environment.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"iSEE/iSEEdemo"</span><span class="op">)</span></span></code></pre></div>
<p>We start by loading the packages that will be used during the
demo.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"SingleCellExperiment"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/iSEE/iSEE" class="external-link">"iSEE"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"TENxPBMCData"</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/iSEE/iSEEu" class="external-link">"iSEEu"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/iSEE/iSEEde" class="external-link">"iSEEde"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/iSEE/iSEEpathways" class="external-link">"iSEEpathways"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/iSEE/iSEEhub" class="external-link">"iSEEhub"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/iSEE/iSEEindex" class="external-link">"iSEEindex"</a></span><span class="op">)</span></span></code></pre></div>
<p>All of these would be loaded if you run</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://iSEE.github.io/iSEEdemo/">"iSEEdemo"</a></span><span class="op">)</span></span></code></pre></div>
<p>You can also clone this repository (<a href="https://iSEE.github.io/iSEEdemo" class="uri">https://iSEE.github.io/iSEEdemo</a>) locally by
<code>git clone</code> - it makes it easy to follow along this
vignette!</p>
</div>
<div class="section level2">
<h2 id="lets-go-get-us-some-data">Let’s go get us some data!<a class="anchor" aria-label="anchor" href="#lets-go-get-us-some-data"></a>
</h2>
<p>We will start very much <em>in medias res</em>, assuming that this
might be the ideal point where you have processed your data - early or
late phase - and now want to have the BEST look at that.</p>
<p>Navigate to <a href="https://zenodo.org/records/10084595" class="external-link uri">https://zenodo.org/records/10084595</a> and download the
data (it contains the three objects you might be “coming from”).<br>
Unpack the archive you downloaded in the <code>datasets</code> subfolder
of the <code>vignette</code> folder.</p>
<p>We included a <code>SingleCellExperiment</code> object on a processed
version for the PBMC3k dataset, many of you can somehow be familiar with
this.</p>
<p>Let’s load the dataset in our environment:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sce_pbmc3k</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span></span>
<span>  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"datasets"</span>, <span class="st">"sce_pbmc3k.RDS"</span>, package <span class="op">=</span> <span class="st">"iSEEdemo"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">sce_pbmc3k</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 32738 2643 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(2): counts logcounts</span></span>
<span><span class="co">#&gt; rownames(32738): MIR1302-10 FAM138A ... AC002321.2 AC002321.1</span></span>
<span><span class="co">#&gt; rowData names(21): ENSEMBL_ID Symbol_TENx ... FDR_cluster13</span></span>
<span><span class="co">#&gt;   FDR_cluster14</span></span>
<span><span class="co">#&gt; colnames(2643): Cell1 Cell2 ... Cell2699 Cell2700</span></span>
<span><span class="co">#&gt; colData names(20): Sample Barcode ... sizeFactor Cluster</span></span>
<span><span class="co">#&gt; reducedDimNames(3): PCA TSNE UMAP</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="as-easy-as-saying-iseesce">As easy as saying <code>iSEE(sce)</code>!<a class="anchor" aria-label="anchor" href="#as-easy-as-saying-iseesce"></a>
</h2>
<p>Who of you is investing a fair amount/lot of time in doing proper
data exploration?</p>
<p>Who of you is familiar with the beautiful CellxGene tool?</p>
<p>Who of you dreamed to do even more on your data?</p>
<p>I have a quiz for you - have a look at this, for example:</p>
<p><a href="http://shiny.imbei.uni-mainz.de:3838/iSEE_Tonsils_CD4Tcells/" class="external-link uri">http://shiny.imbei.uni-mainz.de:3838/iSEE_Tonsils_CD4Tcells/</a></p>
<p><img src="ss_Tcells.png" width="1726"></p>
<details><summary>
Question 1
</summary><p>How many lines of code are required to realize this?</p>
<ol style="list-style-type: upper-alpha">
<li>4-5<br>
</li>
<li>250-300</li>
<li>7500-10000<br>
</li>
</ol></details><details><summary>
Question 2
</summary><p>How many of these lines had to be typed by a human?</p>
<ol style="list-style-type: upper-alpha">
<li>4-5<br>
</li>
<li>250-300<br>
</li>
<li>7500-10000<br>
</li>
</ol></details><p>So, yes, I will show you that is as easy as that:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/iSEE/iSEE" class="external-link">"iSEE"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">sce_pbmc3k</span><span class="op">)</span></span></code></pre></div>
<p>Let’s run this together and see what we have got in store.</p>
<p><img src="iSEEmeme.jpg" align="center" alt="" width="600"></p>
</div>
<div class="section level2">
<h2 id="quick-tour-of-the-panels">Quick tour of the panels<a class="anchor" aria-label="anchor" href="#quick-tour-of-the-panels"></a>
</h2>
<p>What typical visualizations people would you use to represent and
explore your datasets?</p>
<p>Each panel has some essential content:</p>
<ul>
<li>Data parameters</li>
<li>Visual parameters</li>
<li>Selection parameters</li>
</ul>
<p>Let’s have a closer look at that on the Reduced Dimension Plot 1
panel.</p>
<p>If you have no idea of what a panel can “normally” do, look for the
question mark icon, that will start a tiny tour on its
functionality.</p>
</div>
<div class="section level2">
<h2 id="additional-ui-functionality">Additional UI functionality<a class="anchor" aria-label="anchor" href="#additional-ui-functionality"></a>
</h2>
<p>iSEE also has some additional controls, grouped into the header bar,
let’s have a look at them:</p>
<ul>
<li>Organization buttons</li>
<li>Export buttons</li>
<li>Documentation buttons</li>
<li>Extra info buttons</li>
</ul>
<p>With this, we can…</p>
<ul>
<li>reorganize the panel layout - add, remove, re-order</li>
<li>extract the code to exactly reproduce the outputs, 1:1</li>
<li>store (and re-utilize) the panel settings</li>
<li>generate at once an archive with all/selected parts of the panel
outputs</li>
<li>start a tour on the content displayed - yes, you can very much tell
a story with iSEE!</li>
</ul>
</div>
<div class="section level2">
<h2 id="in-depth-exploration-of-data---isees-main-functionality">In depth exploration of data - iSEE’s main functionality<a class="anchor" aria-label="anchor" href="#in-depth-exploration-of-data---isees-main-functionality"></a>
</h2>
<div class="section level3">
<h3 id="you-can-control-pretty-much-everything-in-a-plot">You can control pretty much everything in a plot<a class="anchor" aria-label="anchor" href="#you-can-control-pretty-much-everything-in-a-plot"></a>
</h3>
<p>Coloring by, faceting, zooming, hovering!<br>
Make a wish!</p>
</div>
<div class="section level3">
<h3 id="you-can-link-any-panel-to-any-other-panel">You can link any panel to any other panel<a class="anchor" aria-label="anchor" href="#you-can-link-any-panel-to-any-other-panel"></a>
</h3>
<p>Brush selection, lasso selection, select the receiving panel,
iterate!<br>
Read out information, see the panel transmission graph.</p>
</div>
<div class="section level3">
<h3 id="you-can-use-tables-to-navigate-panels">You can use tables to navigate panels<a class="anchor" aria-label="anchor" href="#you-can-use-tables-to-navigate-panels"></a>
</h3>
<p>Search and select, transmit, receive, iterate!</p>
</div>
<div class="section level3">
<h3 id="configuring-your-isee-instance">Configuring your iSEE instance<a class="anchor" aria-label="anchor" href="#configuring-your-isee-instance"></a>
</h3>
<p>Panels: add, move, remove; exporting the <code>initial</code>
configuration, reloading it</p>
</div>
<div class="section level3">
<h3 id="reproducibility-for-the-win">Reproducibility for the win!<a class="anchor" aria-label="anchor" href="#reproducibility-for-the-win"></a>
</h3>
<p>A magic wand: two clicks away from all the R code you need; export
all outputs, save individual images</p>
</div>
<div class="section level3">
<h3 id="sharing-is-caring-and-so-is-telling-a-story">Sharing is caring, and so is telling a story<a class="anchor" aria-label="anchor" href="#sharing-is-caring-and-so-is-telling-a-story"></a>
</h3>
<p>Start the tour, highlighting elements; make your data really
explorable!</p>
</div>
</div>
<div class="section level2">
<h2 id="custom-panels-the-isee-universe">Custom panels &amp; the iSEE universe<a class="anchor" aria-label="anchor" href="#custom-panels-the-isee-universe"></a>
</h2>
<p>A couple of additional things you can do with custom panels (and this
is just an appetizer!):</p>
<ul>
<li>Custom panels: perform some extra steps</li>
<li>Multiple selection and “DE on the fly”</li>
<li>Check out <em><a href="https://bioconductor.org/packages/3.18/iSEEu" class="external-link">iSEEu</a></em> for
more</li>
<li>Spatial data? We can already handle that with a workaround! See <a href="https://moffittlab.connect.hms.harvard.edu/merfish/merfish_homepage.html" class="external-link uri">https://moffittlab.connect.hms.harvard.edu/merfish/merfish_homepage.html</a>
for an example</li>
</ul>
</div>
<div class="section level2">
<h2 id="got-data">Got data?<a class="anchor" aria-label="anchor" href="#got-data"></a>
</h2>
<div class="section level3">
<h3 id="got-single-cell-data">Got single cell data?<a class="anchor" aria-label="anchor" href="#got-single-cell-data"></a>
</h3>
<p>You might have got the gist:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="va">sce_pbmc3k</span> <span class="co"># Load your data in some manner</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/iSEE/iSEE" class="external-link">"iSEE"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span></code></pre></div>
<p>So, yes, we mighe be down to THREE lines of code.</p>
</div>
<div class="section level3">
<h3 id="got-some-special-feature-in-your-data">Got some special feature in your data?<a class="anchor" aria-label="anchor" href="#got-some-special-feature-in-your-data"></a>
</h3>
<p>Maybe <em><a href="https://bioconductor.org/packages/3.18/iSEEu" class="external-link">iSEEu</a></em> can
be the place to find already that piece of functionality!</p>
</div>
<div class="section level3">
<h3 id="got-bulk">Got bulk?<a class="anchor" aria-label="anchor" href="#got-bulk"></a>
</h3>
<p>Have a look at the functionality provided by</p>
<ul>
<li><em><a href="https://bioconductor.org/packages/3.18/iSEEde" class="external-link">iSEEde</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.18/iSEEpathways" class="external-link">iSEEpathways</a></em></li>
</ul>
<p>These can be ideal companions for exploring DE results.</p>
<p><em><a href="https://bioconductor.org/packages/3.18/iSEEde" class="external-link">iSEEde</a></em> and
<em><a href="https://bioconductor.org/packages/3.18/iSEEpathways" class="external-link">iSEEpathways</a></em>
are two new Bioconductor packages that provide <em><a href="https://bioconductor.org/packages/3.18/iSEE" class="external-link">iSEE</a></em> panels
specifically aimed towards exploration of differential expression and
pathway analysis results. More precisely, <em><a href="https://bioconductor.org/packages/3.18/iSEEde" class="external-link">iSEEde</a></em>
provides the <code>VolcanoPlot</code>, <code>MAPlot</code>,
<code>LogFCLogFCPlot</code> and <code>DETable</code> panels.</p>
</div>
<div class="section level3">
<h3 id="got-many-datasets">Got many datasets?<a class="anchor" aria-label="anchor" href="#got-many-datasets"></a>
</h3>
<p>Have a look at the functionality provided by</p>
<ul>
<li><em><a href="https://bioconductor.org/packages/3.18/iSEEhub" class="external-link">iSEEhub</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.18/iSEEindex" class="external-link">iSEEindex</a></em></li>
</ul>
<p>The <em><a href="https://bioconductor.org/packages/3.18/iSEEhub" class="external-link">iSEEhub</a></em>
package provides a custom landing page for an <em><a href="https://bioconductor.org/packages/3.18/iSEE" class="external-link">iSEE</a></em>
application interfacing with the Bioconductor <em><a href="https://bioconductor.org/packages/3.18/ExperimentHub" class="external-link">ExperimentHub</a></em>.
The landing page allows users to browse the ExperimentHub, select a data
set, download and cache it, and import it directly into an iSEE app.</p>
<p><em><a href="https://bioconductor.org/packages/3.18/iSEEindex" class="external-link">iSEEindex</a></em>
provides an interface to <strong>any</strong> collection of data sets
within a single iSEE web-application.<br>
The main functionality of this package is to define a custom landing
page allowing app maintainers to list a custom collection of data sets
that users can select from and directly load objects into an iSEE web
application.</p>
<blockquote>
<p>One iSEE instance to <del>rule</del> exlore them all…</p>
</blockquote>
<p>A more elaborate example is available at <a href="https://rehwinkellab.shinyapps.io/ifnresource/" class="external-link">https://rehwinkellab.shinyapps.io/ifnresource/</a>.
The source can be found at <a href="https://github.com/kevinrue/IFNresource" class="external-link">https://github.com/kevinrue/IFNresource</a>.</p>
<!-- Possible actions: -->
<!-- navigate to github repo -->
<!-- explore the app.R file -->
<!-- check the initial state files -->
<!-- have a look at the configuration yaml -->
<p>Potential use cases can include:</p>
<ul>
<li>An app to present and explore the different datasets in your next
publication</li>
<li>An app to explore collection of datasets collaboratively, in
consortium-like initiatives</li>
<li>An app to mirror and enhance the content of e.g. the CellxGene data
portal</li>
<li>Got any ideas on how to use iSEE for such deployments?</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="interoperability">Interoperability<a class="anchor" aria-label="anchor" href="#interoperability"></a>
</h2>
<p>“Wait, I got Seurat/Anndata objects”.</p>
<p>The single cell communities got you covered:</p>
<div class="section level3">
<h3 id="from-seurat">From Seurat…<a class="anchor" aria-label="anchor" href="#from-seurat"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://satijalab.org/seurat" class="external-link">"Seurat"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">seurat_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"datasets/seurat_pbmc3k.RDS"</span><span class="op">)</span></span>
<span><span class="va">sce_from_seurat</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/as.SingleCellExperiment.html" class="external-link">as.SingleCellExperiment</a></span><span class="op">(</span></span>
<span>  <span class="va">seurat_object</span></span>
<span><span class="op">)</span></span>
<span><span class="va">sce_from_seurat</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">sce_from_seurat</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="from-anndata">From AnnData…<a class="anchor" aria-label="anchor" href="#from-anndata"></a>
</h3>
<p>… which is basically able to serve any dataset available throught the
CELLxGENE Data Portal!</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/theislab/zellkonverter" class="external-link">"zellkonverter"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">sce_from_anndata</span> <span class="op">&lt;-</span> <span class="fu">zellkonverter</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/zellkonverter/man/readH5AD.html" class="external-link">readH5AD</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"datasets/anndata_pbmc3k.h5ad"</span><span class="op">)</span></span>
<span><span class="va">sce_from_anndata</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">sce_from_anndata</span><span class="op">)</span></span></code></pre></div>
<p>An example could be…</p>
<p><a href="https://cellxgene.cziscience.com/e/c7856243-c59a-4b70-8ce7-25b94c2d9da1.cxg/" class="external-link uri">https://cellxgene.cziscience.com/e/c7856243-c59a-4b70-8ce7-25b94c2d9da1.cxg/</a></p>
</div>
</div>
<div class="section level2">
<h2 id="i-want-moaaaaar-on-isee">I want moaaaaar on iSEE!<a class="anchor" aria-label="anchor" href="#i-want-moaaaaar-on-isee"></a>
</h2>
<p>For more extensive descriptions of the different parts of the
interface, we refer to the <a href="https://isee.github.io/iSEEWorkshopEuroBioc2020/articles/overview.html" class="external-link">overview</a>
vignette in a previous, extended workshop on <em><a href="https://bioconductor.org/packages/3.18/iSEE" class="external-link">iSEE</a></em>.</p>
<p>Additional material can also be found at these other
repositories:</p>
<ul>
<li><a href="https://isee.github.io/iSEEWorkshop2020/" class="external-link uri">https://isee.github.io/iSEEWorkshop2020/</a></li>
<li><a href="https://isee.github.io/iSEEWorkshopEuroBioc2020/" class="external-link uri">https://isee.github.io/iSEEWorkshopEuroBioc2020/</a></li>
<li><a href="https://isee.github.io/iSEEDemoEuroBioC2023/" class="external-link uri">https://isee.github.io/iSEEDemoEuroBioC2023/</a></li>
</ul>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">#&gt; Platform: x86_64-apple-darwin20 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: macOS Monterey 12.7.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/Berlin</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats    graphics utils    stats4   methods  base    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] iSEEdemo_1.0.0              iSEEindex_1.0.0            </span></span>
<span><span class="co">#&gt;  [3] iSEEhub_1.4.0               ExperimentHub_2.10.0       </span></span>
<span><span class="co">#&gt;  [5] AnnotationHub_3.10.0        BiocFileCache_2.10.1       </span></span>
<span><span class="co">#&gt;  [7] dbplyr_2.4.0                iSEEpathways_1.0.0         </span></span>
<span><span class="co">#&gt;  [9] iSEEde_1.0.0                iSEEu_1.14.0               </span></span>
<span><span class="co">#&gt; [11] iSEEhex_1.4.0               TENxPBMCData_1.20.0        </span></span>
<span><span class="co">#&gt; [13] HDF5Array_1.30.0            rhdf5_2.46.0               </span></span>
<span><span class="co">#&gt; [15] DelayedArray_0.28.0         SparseArray_1.2.2          </span></span>
<span><span class="co">#&gt; [17] S4Arrays_1.2.0              abind_1.4-5                </span></span>
<span><span class="co">#&gt; [19] Matrix_1.6-2                iSEE_2.14.0                </span></span>
<span><span class="co">#&gt; [21] SingleCellExperiment_1.24.0 SummarizedExperiment_1.32.0</span></span>
<span><span class="co">#&gt; [23] Biobase_2.62.0              GenomicRanges_1.54.1       </span></span>
<span><span class="co">#&gt; [25] GenomeInfoDb_1.38.1         IRanges_2.36.0             </span></span>
<span><span class="co">#&gt; [27] S4Vectors_0.40.1            BiocGenerics_0.48.1        </span></span>
<span><span class="co">#&gt; [29] MatrixGenerics_1.14.0       matrixStats_1.1.0          </span></span>
<span><span class="co">#&gt; [31] BiocStyle_2.30.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RColorBrewer_1.1-3            rstudioapi_0.15.0            </span></span>
<span><span class="co">#&gt;   [3] jsonlite_1.8.7                shape_1.4.6                  </span></span>
<span><span class="co">#&gt;   [5] magrittr_2.0.3                rmarkdown_2.25               </span></span>
<span><span class="co">#&gt;   [7] GlobalOptions_0.1.2           fs_1.6.3                     </span></span>
<span><span class="co">#&gt;   [9] zlibbioc_1.48.0               ragg_1.2.6                   </span></span>
<span><span class="co">#&gt;  [11] vctrs_0.6.4                   DelayedMatrixStats_1.24.0    </span></span>
<span><span class="co">#&gt;  [13] memoise_2.0.1                 RCurl_1.98-1.13              </span></span>
<span><span class="co">#&gt;  [15] htmltools_0.5.7               curl_5.1.0                   </span></span>
<span><span class="co">#&gt;  [17] Rhdf5lib_1.24.0               sass_0.4.7                   </span></span>
<span><span class="co">#&gt;  [19] bslib_0.5.1                   htmlwidgets_1.6.2            </span></span>
<span><span class="co">#&gt;  [21] desc_1.4.2                    cachem_1.0.8                 </span></span>
<span><span class="co">#&gt;  [23] igraph_1.5.1                  mime_0.12                    </span></span>
<span><span class="co">#&gt;  [25] lifecycle_1.0.4               iterators_1.0.14             </span></span>
<span><span class="co">#&gt;  [27] pkgconfig_2.0.3               colourpicker_1.3.0           </span></span>
<span><span class="co">#&gt;  [29] R6_2.5.1                      fastmap_1.1.1                </span></span>
<span><span class="co">#&gt;  [31] GenomeInfoDbData_1.2.11       shiny_1.7.5.1                </span></span>
<span><span class="co">#&gt;  [33] clue_0.3-65                   digest_0.6.33                </span></span>
<span><span class="co">#&gt;  [35] colorspace_2.1-0              paws.storage_0.4.0           </span></span>
<span><span class="co">#&gt;  [37] AnnotationDbi_1.64.1          DESeq2_1.42.0                </span></span>
<span><span class="co">#&gt;  [39] rprojroot_2.0.4               textshaping_0.3.7            </span></span>
<span><span class="co">#&gt;  [41] RSQLite_2.3.3                 beachmat_2.18.0              </span></span>
<span><span class="co">#&gt;  [43] filelock_1.0.2                urltools_1.7.3               </span></span>
<span><span class="co">#&gt;  [45] fansi_1.0.5                   httr_1.4.7                   </span></span>
<span><span class="co">#&gt;  [47] mgcv_1.9-0                    compiler_4.3.0               </span></span>
<span><span class="co">#&gt;  [49] bit64_4.0.5                   doParallel_1.0.17            </span></span>
<span><span class="co">#&gt;  [51] BiocParallel_1.36.0           DBI_1.1.3                    </span></span>
<span><span class="co">#&gt;  [53] shinyAce_0.4.2                highr_0.10                   </span></span>
<span><span class="co">#&gt;  [55] hexbin_1.28.3                 rappdirs_0.3.3               </span></span>
<span><span class="co">#&gt;  [57] rjson_0.2.21                  tools_4.3.0                  </span></span>
<span><span class="co">#&gt;  [59] vipor_0.4.5                   interactiveDisplayBase_1.40.0</span></span>
<span><span class="co">#&gt;  [61] httpuv_1.6.12                 glue_1.6.2                   </span></span>
<span><span class="co">#&gt;  [63] nlme_3.1-163                  rhdf5filters_1.14.1          </span></span>
<span><span class="co">#&gt;  [65] promises_1.2.1                grid_4.3.0                   </span></span>
<span><span class="co">#&gt;  [67] cluster_2.1.4                 generics_0.1.3               </span></span>
<span><span class="co">#&gt;  [69] gtable_0.3.4                  utf8_1.2.4                   </span></span>
<span><span class="co">#&gt;  [71] XVector_0.42.0                ggrepel_0.9.4                </span></span>
<span><span class="co">#&gt;  [73] BiocVersion_3.18.0            foreach_1.5.2                </span></span>
<span><span class="co">#&gt;  [75] pillar_1.9.0                  stringr_1.5.0                </span></span>
<span><span class="co">#&gt;  [77] limma_3.58.1                  later_1.3.1                  </span></span>
<span><span class="co">#&gt;  [79] rintrojs_0.3.3                circlize_0.4.15              </span></span>
<span><span class="co">#&gt;  [81] splines_4.3.0                 dplyr_1.1.3                  </span></span>
<span><span class="co">#&gt;  [83] lattice_0.22-5                bit_4.0.5                    </span></span>
<span><span class="co">#&gt;  [85] paws.common_0.6.4             tidyselect_1.2.0             </span></span>
<span><span class="co">#&gt;  [87] locfit_1.5-9.8                ComplexHeatmap_2.18.0        </span></span>
<span><span class="co">#&gt;  [89] scuttle_1.12.0                Biostrings_2.70.1            </span></span>
<span><span class="co">#&gt;  [91] miniUI_0.1.1.1                knitr_1.45                   </span></span>
<span><span class="co">#&gt;  [93] edgeR_4.0.1                   xfun_0.41                    </span></span>
<span><span class="co">#&gt;  [95] shinydashboard_0.7.2          datasets_4.3.0               </span></span>
<span><span class="co">#&gt;  [97] statmod_1.5.0                 DT_0.30                      </span></span>
<span><span class="co">#&gt;  [99] stringi_1.8.1                 yaml_2.3.7                   </span></span>
<span><span class="co">#&gt; [101] shinyWidgets_0.8.0            evaluate_0.23                </span></span>
<span><span class="co">#&gt; [103] codetools_0.2-19              tibble_3.2.1                 </span></span>
<span><span class="co">#&gt; [105] BiocManager_1.30.22           cli_3.6.1                    </span></span>
<span><span class="co">#&gt; [107] xtable_1.8-4                  systemfonts_1.0.5            </span></span>
<span><span class="co">#&gt; [109] munsell_0.5.0                 jquerylib_0.1.4              </span></span>
<span><span class="co">#&gt; [111] Rcpp_1.0.11                   triebeard_0.4.1              </span></span>
<span><span class="co">#&gt; [113] png_0.1-8                     parallel_4.3.0               </span></span>
<span><span class="co">#&gt; [115] ellipsis_0.3.2                pkgdown_2.0.7                </span></span>
<span><span class="co">#&gt; [117] ggplot2_3.4.4                 blob_1.2.4                   </span></span>
<span><span class="co">#&gt; [119] sparseMatrixStats_1.14.0      bitops_1.0-7                 </span></span>
<span><span class="co">#&gt; [121] viridisLite_0.4.2             scales_1.2.1                 </span></span>
<span><span class="co">#&gt; [123] grDevices_4.3.0               purrr_1.0.2                  </span></span>
<span><span class="co">#&gt; [125] crayon_1.5.2                  GetoptLong_1.0.5             </span></span>
<span><span class="co">#&gt; [127] rlang_1.1.2                   KEGGREST_1.42.0              </span></span>
<span><span class="co">#&gt; [129] shinyjs_2.1.0</span></span></code></pre></div>
</div>
<div class="footnotes footnotes-end-of-document">
<hr>
<ol>
<li id="fn1"><p><a href="mailto:marinif@uni-mainz.de" class="email">marinif@uni-mainz.de</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p><a href="mailto:charlottesoneson@gmail.com" class="email">charlottesoneson@gmail.com</a><a href="#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p><a href="mailto:kevinrue67@gmail.com" class="email">kevinrue67@gmail.com</a><a href="#fnref3" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Federico Marini, Charlotte Soneson, Kevin Rue-Albrecht.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
